### PR TITLE
Add an exponential backoff policy to the retrier

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Implement an exponential backoff policy in the `retrier`.

In case of failure from the agent to connect to docker, the kubernetes API server, the kubelet, etc., it used to retry every 30 seconds 10 times and then, to give up.
The agent will now retry after 1 second. It will then double the period between two consecutive retries each time, up to 5 minutes.
So, After 10 minutes, the agent will keep on retrying every 5 minutes instead of completely giving up after 5 minutes.    

### Motivation

This change will avoid to have to restart the agent if it started in an environment that remains degraded for a while (docker being down for several minutes for example).

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by sunhay — APPROVED** [review]
LGTM for Networks

**Review by clamoriniere — APPROVED** [review]
💯

File: pkg/util/retry/retrier.go [hunk 1]
```diff
@@ -110,6 +117,15 @@ func (r *Retrier) doTry() *Error {
 				r.status = FailWillRetry
 				r.nextTry = time.Now().Add(r.cfg.RetryDelay - 100*time.Millisecond)
 			}
+		case Backoff:
+			sleep := r.cfg.InitialRetryDelay * 1 << r.tryCount
```
Comment (ajacquemot) [inline pkg/util/retry/retrier.go]: noob question, what happens if we remove the `* 1` ? Wrong type ?

**Review by ajacquemot — APPROVED** [review]

**Review by prognant — APPROVED** [review]
LGTM, minor remark however as a backoff strategy could be linear or linear/exponential mod(x) instead of pure exponential, the variable naming may be adjusted. Especially the retry strategy name that could have been ExponentialBackoff instead of just Backoff.

**Pull request merged**